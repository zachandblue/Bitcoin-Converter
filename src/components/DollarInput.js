import React, { Component } from "react";

export class DollarInput extends Component {
  constructor() {
    super();

    this.state = {
      dollars: ""
    };
  }
  render() {
    return (
      <div className="dollar-input">
        <form>
          <div className="dollars">
            <h3>US Dollars</h3>
          </div>
          <input
            type="text"
            placeholder="$"
            className="dollar-input"
            onChange={this.props.handleChange}
            value={this.props.dollars}
          />
        </form>
      </div>
    );
  }
}

export default DollarInput;
