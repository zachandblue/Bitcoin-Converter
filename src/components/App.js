import React, { Component } from "react";
import DollarInput from "./DollarInput";
import Bitcoin from "./Bitcoin";
import getData from "../api";

export class App extends Component {
  constructor() {
    super();

    this.state = {
      dollars: "",
      exchangeRate: "",
      bitcoins: ""
    };

    this.apiCall = this.apiCall.bind(this);
  }

  handleChange = e => {
    const dollars = e.target.value;
    this.setState({ dollars });
  };

  apiCall = async () => {
    const result = await getData();
    const exchangeRate = result.bpi.USD.rate_float;
    this.setState({ exchangeRate });
    return true;
  };

  componentDidMount = () => {
    this.apiCall();
  };
  render() {
    return (
      <div className="main">
        <div className="header">
          <h1>Cash To Bitcoin</h1>
          <i className="fab fa-bitcoin" />
          <div className="header-triangle-1" />
          <div className="header-triangle-2" />
        </div>

        <div className="app">
          <div className=" main-box-1" />
          <DollarInput
            handleChange={this.handleChange}
            dollars={this.state.dollars}
          />

          <Bitcoin
            exchangeRate={this.state.exchangeRate}
            dollars={this.state.dollars}
            exchangeRate={this.state.exchangeRate}
          />
          <div className=" main-box-2" />
          <div className=" main-box-3" />
          <div className=" main-box-4" />
          <div className=" main-box-5" />
          <div className=" main-box-6" />
          <div className="credits">
            Powered by <a href="http://www.coindesk.com/price">CoinDesk</a>
          </div>
        </div>

        <div className="footer">
          {/* <div className="footer-triangle-1" /> */}
        </div>
      </div>
    );
  }
}

export default App;
