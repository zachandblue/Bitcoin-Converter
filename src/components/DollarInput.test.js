import React from "react";
import { shallow, mount } from "enzyme";
import { DollarInput } from "./DollarInput";
import { App } from "./App";

describe("DollarImport", () => {
  const dollarInput = shallow(<DollarInput />);
  it("renders correctly", () => {
    expect(dollarInput).toMatchSnapshot();
  });
  it("creates an input for dollars", () => {
    expect(dollarInput.find(".dollar-input").exists()).toBe(true);
  });
});
