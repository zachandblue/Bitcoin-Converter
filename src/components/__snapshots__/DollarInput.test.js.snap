// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DollarImport renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DollarInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
                className="dollars"
        >
                <h3>
                        US Dollars
                </h3>
        </div>
        <input
                className="dollar-input"
                onChange={undefined}
                placeholder="$"
                type="text"
                value={undefined}
        />
</form>,
      "className": "dollar-input",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dollars"
>
            <h3>
                        US Dollars
            </h3>
</div>,
          <input
            className="dollar-input"
            onChange={undefined}
            placeholder="$"
            type="text"
            value={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              US Dollars
</h3>,
            "className": "dollars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "US Dollars",
            },
            "ref": null,
            "rendered": "US Dollars",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dollar-input",
            "onChange": undefined,
            "placeholder": "$",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
                    className="dollars"
          >
                    <h3>
                              US Dollars
                    </h3>
          </div>
          <input
                    className="dollar-input"
                    onChange={undefined}
                    placeholder="$"
                    type="text"
                    value={undefined}
          />
</form>,
        "className": "dollar-input",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dollars"
>
              <h3>
                            US Dollars
              </h3>
</div>,
            <input
              className="dollar-input"
              onChange={undefined}
              placeholder="$"
              type="text"
              value={undefined}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                US Dollars
</h3>,
              "className": "dollars",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "US Dollars",
              },
              "ref": null,
              "rendered": "US Dollars",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dollar-input",
              "onChange": undefined,
              "placeholder": "$",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
